scripts/lib/CIME/case.py

```diff
@@ -116,9 +116,9 @@ def check_if_comp_var(self, vid):
         comp = None
         iscompvar = False
         for env_file in self._env_entryid_files:
-            vid, comp, iscompvar = env_file.check_if_comp_var(vid)
+            new_vid, new_comp, iscompvar = env_file.check_if_comp_var(vid)
```

Comment:
Wouldn't it be better to change this inside check_if_comp_var so that the original vid is returned if iscompvar is False?

Reply:
Yeah, that's probably better.

Reply:
```
diff --git a/scripts/lib/CIME/XML/env_base.py b/scripts/lib/CIME/XML/env_base.py
index 1540139c1..1622adbc3 100644
--- a/scripts/lib/CIME/XML/env_base.py
+++ b/scripts/lib/CIME/XML/env_base.py
@@ -46,13 +46,13 @@ class EnvBase(EntryID):
 
         for comp in self._components:
             if "_"+comp in vid:
-                vid = string.replace(vid, '_'+comp, '', 1)
+                new_vid = string.replace(vid, '_'+comp, '', 1)
                 break
             elif comp+"_" in vid:
-                vid = string.replace(vid, comp+'_', '', 1)
+                new_vid = string.replace(vid, comp+'_', '', 1)
                 break
-        if vid in self._component_value_list:
-            return vid, comp, True
+        if new_vid in self._component_value_list:
+            return new_vid, comp, True
         return vid, None, False
```